### PR TITLE
gnrc: make message queues static

### PR DESCRIPTION
**Pull request by fabian18**
### Contribution description

I saw messages sent to the IPv6 thread being dropped.
I think it could happen that an address does not become valid for that reason.
So I increased `CONFIG_GNRC_IPV6_MSG_QUEUE_SIZE_EXP` and a stack overflow happened.
This is not supposed to happen for a `CONFIG_` variable. 

### Testing procedure

- internal application

```
2023-10-19 17:19:17,118 # queue_msg(): Cannot send message 0x2000a7b0 to message queue of thread 4 is full (or there is none)
2023-10-19 17:19:17,148 # queue_msg(): Cannot send message 0x2000a7b0 to message queue of thread 4 is full (or there is none)
2023-10-19 17:19:17,149 # queue_msg(): Cannot send message 0x2000a7b0 to message queue of thread 4 is full (or there is none)
2023-10-19 17:19:17,422 # queue_msg(): Cannot send message 0x2000a7b0 to message queue of thread 4 is full (or there is none)
2023-10-19 17:19:17,438 # queue_msg(): Cannot send message 0x2000a7b0 to message queue of thread 4 is full (or there is none)
2023-10-19 17:19:17,693 # queue_msg(): Cannot send message 0x2000a7b0 to message queue of thread 4 is full (or there is none)
2023-10-19 17:19:17,724 # queue_msg(): Cannot send message 0x2000a7b0 to message queue of thread 4 is full (or there is none)
2023-10-19 17:19:17,950 # queue_msg(): Cannot send message 0x2000a7b0 to message queue of thread 4 is full (or there is none)
2023-10-19 17:19:18,029 # queue_msg(): Cannot send message 0x2000a7b0 to message queue of thread 4 is full (or there is none)
2023-10-19 17:19:18,318 # queue_msg(): Cannot send message 0x2000a7b0 to message queue of thread 4 is full (or there is none)
2023-10-19 17:19:18,332 # queue_msg(): Cannot send message 0x2000a7b0 to message queue of thread 4 is full (or there is none)
2023-10-19 17:19:18,589 # queue_msg(): Cannot send message 0x2000a7b0 to message queue of thread 4 is full (or there is none)
2023-10-19 17:19:18,590 # queue_msg(): Cannot send message 0x2000a7b0 to message queue of thread 4 is full (or there is none)
2023-10-19 17:19:18,798 # queue_msg(): Cannot send message 0x20008888 to message queue of thread 4 is full (or there is none)
```

```
2023-10-19 17:32:33,253 # ps
2023-10-19 17:32:33,270 # 	pid | name                 | state    Q | pri | stack  ( used) ( free) | base addr  | current     
2023-10-19 17:32:33,272 # 	  - | isr_stack            | -        - |   - |   2048 (  492) ( 1556) | 0x20000000 | 0x200007c8
2023-10-19 17:32:33,287 # 	  2 | pktdump              | bl rx    _ |   6 |   5120 (  240) ( 4880) | 0x2001b0ec | 0x2001c3fc 
2023-10-19 17:32:33,301 # 	  3 | 6lo                  | bl rx    _ |   3 |   1152 (  416) (  736) | 0x2001c74c | 0x2001ca44 
2023-10-19 17:32:33,303 # 	  4 | ipv6                 | bl rx    _ |   4 |   1152 (  844) (  308) | 0x2000854c | 0x2000883c 
2023-10-19 17:32:33,318 # 	  5 | udp                  | bl rx    _ |   5 |    576 (  312) (  264) | 0x2001cbcc | 0x2001cd0c 

```

### CODE REVIEW

**Comment by miri64** [comment]
Mhh... to what did you increase the queue size that it overflowed the stack? Does it make sense to increase the stack size as well then?

The problem with adding the queue to static memory for _all_ modules is that this effectively increases the RAM usage of GNRC. Does it make sense to this only for IPv6 for now?

File: core/msg.c [hunk 1]
```diff
@@ -46,7 +46,8 @@ static int queue_msg(thread_t *target, const msg_t *m)
     int n = cib_put(&(target->msg_queue));
 
     if (n < 0) {
-        DEBUG("queue_msg(): message queue is full (or there is none)\n");
+        DEBUG("queue_msg(): message queue of thread %"PRIkernel_pid
+              " is full (or there is none)\n", target->pid);
```
Comment (miri64) [inline core/msg.c:50]: Uncrustify says:
```suggestion
        DEBUG("queue_msg(): message queue of thread %" PRIkernel_pid
              " is full (or there is none)\n", target->pid);
```

**Comment by fabian18** [comment]
> Mhh... to what did you increase the queue size that it overflowed the stack? Does it make sense to increase the stack size as well then?

I increased the size of the IPv6 message queue to 2⁶ to make sure that my problem of an address not getting VALID sometimes is related to that. I don´t want to increase the size upstream. I only want to  give the opportunity. 
I would like it if stack size and message queue size are independent. I would say stack size scales with code complexity and message queue size scales with IPC communication. Since NIB is posting messages for every interface to IPv6 for doing DAD, router/neighbor discovery, there is a lot of IPC communication.

> The problem with adding the queue to static memory for all modules is that this effectively increases the RAM usage of GNRC. Does it make sense to do this only for IPv6 for now?

Having the queue not static and configurable is a bad pattern in my eyes.
Given `sizeof(msg_t) == 8` and someone used 6LO, IPv6, UDP, each one having a default queue size of 8, it increases RAM usage by 192 bytes.

> You should then reduce the stack size of all GNRC threads by the default queue size to avoid increasing the memory requirements.

```C
#ifndef GNRC_IPV6_STACK_SIZE
#define GNRC_IPV6_STACK_SIZE           (THREAD_STACKSIZE_DEFAULT)
#endif

#ifndef GNRC_PKTDUMP_STACKSIZE
#define GNRC_PKTDUMP_STACKSIZE         (THREAD_STACKSIZE_MAIN)
#endif

#ifndef GNRC_UDP_STACK_SIZE
#define GNRC_UDP_STACK_SIZE            (THREAD_STACKSIZE_SMALL)
#endif

#ifndef GNRC_SIXLOWPAN_STACK_SIZE
#define GNRC_SIXLOWPAN_STACK_SIZE      (THREAD_STACKSIZE_DEFAULT)
#endif
```

Just subtract 64 bytes there, or is there anything better? If you want, yes but it does not look so nice and someone may wonder why.

**Comment by benpicco** [comment]
I think subtracting 64 bytes from each of those defines is not too bad, I ended up doing the same for [`GNRC_NETIF_STACKSIZE_DEFAULT`](https://github.com/RIOT-OS/RIOT/blob/master/sys/net/gnrc/netif/init_devs/include/init_devs.h#L38) for the same reason.

We are lucky that the size of `msg_t` is architecture independent.

**Review by benpicco — APPROVED** [review]
Looks good to me!

**Pull request merged**